system/Commands/Utilities/Routes/AutoRouterImproved/ControllerMethodReader.php

```diff
@@ -93,6 +93,11 @@ public function read(string $class, string $defaultController = 'Home', string $
                         continue;
                     }
 
+                    // Skip the default controller.
+                    if (class_basename($class) === $defaultController) {
+                        continue;
+                    }
```

Comment:
just quick read, `$class` and `$defaultController`seems never replaced under loop, seems can be moved before loop to return  early.

Reply:
Thank you for review. But it cannot be moved before loop.
The method is a bit difficult to understand.
I refactored and and some comments.

Reply:
@samsonasik I changed my mind. Please review #7419 when you have time.